lib/compat.php

```diff
@@ -125,3 +125,86 @@ function gutenberg_get_post_from_context() {
 	}
 	return get_post();
 }
+
+/**
+ * Shim that hooks into `pre_render_block` so as to override `render_block` with
+ * a function that assigns block context.
+ *
+ * This can be removed when plugin support requires WordPress 5.5.0+.
+ *
+ * @see (TBD Trac Link)
+ *
+ * @param string $pre_render The pre-rendered content. Defaults to null.
+ * @param array  $next_block The block being rendered.
+ *
+ * @return string String of rendered HTML.
+ */
+function gutenberg_provide_render_callback_with_block_object( $pre_render, $next_block ) {
+	global $post, $block, $_block_context;
+
+	$source_block = $next_block;
+
+	/** This filter is documented in src/wp-includes/blocks.php */
+	$block = apply_filters( 'render_block_data', $next_block, $source_block );
+
+	$block_type    = WP_Block_Type_Registry::get_instance()->get_registered( $block['blockName'] );
+	$is_dynamic    = $block['blockName'] && null !== $block_type && $block_type->is_dynamic();
+	$block_content = '';
+	$index         = 0;
+
+	if ( ! is_array( $block['attrs'] ) ) {
+		$block['attrs'] = array();
+	}
+
+	if ( ! isset( $block['context'] ) ) {
+		$block['context'] = array();
+	}
+
+	$block_context_before = $_block_context;
+
+	if ( ! isset( $_block_context ) ) {
+		$_block_context = array();
+	}
+
+	if ( ! isset( $_block_context['postId'] ) || $post->ID !== $_block_context['postId'] ) {
+		$_block_context['postId'] = $post->ID;
+	}
+
+	if ( ! empty( $block_type->providesContext ) && is_array( $block_type->providesContext ) ) {
+		if ( ! isset( $_block_context ) ) {
+			$_block_context = array();
+		}
```

Comment:
> This seems to do the same as line 165-167.

Good catch. It was a bad rebase. I had decided at one point to change this so that the global is always assigned. It was related to some of the same thinking about always providing a value to a block: If we're only assigning the global at the point that `providesContext` is determined, it could lead to errors if a filter or block implementation assumes that the global exists. Moving it up to the logic above gives a better guarantee that it is.

tl;dr This should have been removed.

Reply:
> This seems to do the same as line 165-167.

Removed in 88fc9116f.